testsuite/MDAnalysisTests/topology/test_fetch_pdb.py

```diff
@@ -0,0 +1,125 @@
+#
+
+from urllib import request
+
+import MDAnalysis as mda
+import pytest
+
+
+def has_pooch():
+    try:
+        import pooch
+        from requests.exceptions import HTTPError
+
+        return True
+    except ModuleNotFoundError:
+        return False
+
+
+def has_internet():
+    try:
+        request.urlopen("https://files.wwpdb.org/", timeout=2)
+        return True
+    except request.URLError:
+        return False
+
+
+@pytest.mark.skipif(
+    not has_pooch() or not has_internet(),
+    reason="Cannot connect to https://files.wwpdb.org/",
+)
+class TestDocstringExamples:
+    """This class tests all the examples found in fetch_pdb's docstring"""
+
+    TRUE_NATOMS_PER_PDB = {"1AKE": 3816, "4BWZ": 2824}
+
+    def test_one_file_download(self, tmp_path):
+        assert isinstance(
+            mda.fetch_pdb("1AKE", cache_path=tmp_path, file_format="cif"), str
+        )
+
+    def test_multiple_files_download(self, tmp_path):
+        list_of_path_strings = mda.fetch_pdb(
+            ["1AKE", "4BWZ"], cache_path=tmp_path, progressbar=True
+        )
+        assert all(isinstance(PDB_ID, str) for PDB_ID in list_of_path_strings)
+
+    def test_one_file_to_universe(self, tmp_path):
+        u = mda.Universe(
+            mda.fetch_pdb("1AKE"),
+            file_format="pdb.gz",
+            cache_path=tmp_path,
+            progressbar=True,
+        )
+        assert isinstance(u, mda.Universe) and (
+            len(u.atoms) == self.TRUE_NATOMS_PER_PDB["1AKE"]
+        )
+
+    def test_multiple_files_to_universe(self, tmp_path):
+        list_of_path_strings = [
+            mda.Universe(
+                mda.fetch_pdb(PDB_ID),
+                cache_path=tmp_path,
+                file_format="pdb.gz",
+            )
+            for PDB_ID in ("1AKE", "4BWZ")
+        ]
+        assert (
+            all(
+                isinstance(PDB_ID, mda.Universe)
+                for PDB_ID in list_of_path_strings
+            )
+        ) and (
+            len(u.atoms) == n_atoms
+            for u, n_atoms in zip(
+                list_of_path_strings, self.TRUE_NATOMS_PER_PDB.values()
+            )
+        )
+
+
+@pytest.mark.skipif(
+    not has_pooch() or not has_internet(),
+    reason="Cannot connect to https://files.wwpdb.org/",
+)
+class TestExpectedErrors:
+    from requests.exceptions import HTTPError
+
+    def test_invalid_pdb(self, tmp_path):
+        with pytest.raises(self.HTTPError):
+            mda.fetch_pdb(PDB_IDS="foobar", cache_path=tmp_path)
+
+    def test_invalid_file_format(self, tmp_path):
+        with pytest.raises(self.HTTPError):
+            mda.fetch_pdb(
+                PDB_IDS="1AKE", cache_path=tmp_path, file_format="barfoo"
+            )
+
+
+@pytest.mark.skipif(
+    has_pooch(),
+    reason="Pooch is installed.",
+)
+def test_pooch_installation(tmp_path):
+    with pytest.raises(ModuleNotFoundError):
+        mda.fetch_pdb("1AKE", cache_path=tmp_path, file_format="cif")
```

Comment:
Write the test so that it always runs. Look at other examples where we temporary remove an imported library from the path to test behavior in the absence of the module (eg distopia, gsd, ... you can use `git grep gsd` to find files where something is used).